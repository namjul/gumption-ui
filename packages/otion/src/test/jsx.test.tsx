/** @jsx OtionStyling.jsx */

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import * as OtionStyling from '..';

afterEach(cleanup);

describe('otion', () => {
  test('renders', () => {
    const { asFragment } = render(<div>Hello</div>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('render `style` attribute', () => {
    const { asFragment } = render(<div style={{ color: 'red' }}>Hello</div>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('render with `css` prop', () => {
    const { asFragment } = render(<div css={{ color: 'red' }}>Hello</div>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('render with merged className', () => {
    const { asFragment } = render(
      <div css={{ color: 'red' }} className="satori">
        Hello
      </div>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('returns responsive interpolated styles', () => {
    const { asFragment } = render(
      <div
        css={{
          color: 'primary',
          padding: ['small', 'medium', 'large'],
          margin: [undefined, 'medium', undefined],
          ':hover': [{ width: 'small' }, { width: 'medium' }],
        }}
        className="satori"
      >
        Responsive stles
      </div>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('functional values can return responsive arrays', () => {
    const { asFragment } = render(
      <div
        css={{
          color: (t) => [t.scales?.colors.primary, t.scales?.colors.secondary],
        }}
        className="satori"
      >
        Responsive styles
      </div>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('returns selectors interpolated styles', () => {
    const { asFragment } = render(
      <div
        css={{
          selectors: {
            '& > * + *': {
              color: 'primary',
            },
            '&:focus, &:active': {
              color: 'primary',
            },
          },
        }}
        className="satori"
      >
        selectors styles
      </div>,
    );
    expect(asFragment()).toMatchSnapshot();

    // const result = interpolate({
    //   selectors: {
    //     '& > * + *': {
    //       color: 'primary',
    //     },
    //     '&:focus, &:active': {
    //       color: 'primary',
    //     },
    //   },
    // })({ theme });
    // expect(result).toEqual({
    //   selectors: {
    //     '& > * + *': {
    //       color: 'tomato',
    //     },
    //     '&:focus, &:active': {
    //       color: 'tomato',
    //     },
    //   },
    // });
  });

  // test('returns responsive selectors interpolated styles', () => {
  //   const result = interpolate({
  //     selectors: [
  //       {
  //         '& > * + *': {
  //           color: 'primary',
  //         },
  //         '&:focus, &:active': {
  //           color: 'primary',
  //         },
  //       },
  //       {
  //         '& > * + *': {
  //           color: 'secondary',
  //         },
  //         '&:focus, &:active': {
  //           color: 'secondary',
  //         },
  //       },
  //     ],
  //   })({ theme });
  //   expect(result).toEqual({
  //     selectors: {
  //       '& > * + *': {
  //         color: 'tomato',
  //       },
  //       '&:focus, &:active': {
  //         color: 'tomato',
  //       },
  //     },
  //     '@media': {
  //       '(min-width: 640px)': {
  //         selectors: {
  //           '& > * + *': {
  //             color: 'cyan',
  //           },
  //           '&:focus, &:active': {
  //             color: 'cyan',
  //           },
  //         },
  //       },
  //     },
  //   });
  // });

  // test('returns at-rule interpolated styles', () => {
  //   const result = interpolate({
  //     '@media': {
  //       '(min-width: 600px)': {
  //         color: 'primary',
  //       },
  //     },
  //     '@supports': {
  //       '(display: grid)': {
  //         color: 'primary',
  //       },
  //     },
  //   })(theme);
  //   expect(result).toEqual({
  //     '@media': {
  //       '(min-width: 600px)': {
  //         color: 'tomato',
  //       },
  //     },
  //     '@supports': {
  //       '(display: grid)': {
  //         color: 'tomato',
  //       },
  //     },
  //   });
  // });

  //   test('returns at-rule interpolated styles', () => {
  //     const result = interpolate({
  //       '@media': {
  //         '(min-width: 600px)': {
  //           ':hover': {
  //             color: 'primary',
  //           },
  //         },
  //       },
  //       '@supports': {
  //         '(display: grid)': {
  //           ':hover': {
  //             color: 'primary',
  //           },
  //         },
  //       },
  //     })(theme);
  //     expect(result).toEqual({
  //       '@media': {
  //         '(min-width: 600px)': {
  //           ':hover': {
  //             color: 'tomato',
  //           },
  //         },
  //       },
  //       '@supports': {
  //         '(display: grid)': {
  //           ':hover': {
  //             color: 'tomato',
  //           },
  //         },
  //       },
  //     });
  //   });
});
