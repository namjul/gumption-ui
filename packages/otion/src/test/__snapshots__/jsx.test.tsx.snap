// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`otion functional values can return responsive arrays 1`] = `
<DocumentFragment>
  <div
    class="satori"
  >
    Responsive styles
  </div>
</DocumentFragment>
`;

exports[`otion render \`style\` attribute 1`] = `
<DocumentFragment>
  <div
    style="color: red;"
  >
    Hello
  </div>
</DocumentFragment>
`;

exports[`otion render with \`css\` prop 1`] = `
<DocumentFragment>
  <div
    class="_hwfcu5"
  >
    Hello
  </div>
</DocumentFragment>
`;

exports[`otion render with merged className 1`] = `
<DocumentFragment>
  <div
    class="satori _hwfcu5"
  >
    Hello
  </div>
</DocumentFragment>
`;

exports[`otion renders 1`] = `
<DocumentFragment>
  <div>
    Hello
  </div>
</DocumentFragment>
`;

exports[`otion returns responsive interpolated styles 1`] = `
<DocumentFragment>
  <div
    class="satori _i8l9rb _1o85aql _1ns85tw"
  >
    Responsive stles
  </div>
</DocumentFragment>
`;

exports[`otion returns selectors interpolated styles 1`] = `
<DocumentFragment>
  <div
    class="satori _vz9a6n _1mbad9m _19wb1re"
  >
    selectors styles
  </div>
</DocumentFragment>
`;
